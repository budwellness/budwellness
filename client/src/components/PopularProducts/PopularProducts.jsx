import React from 'react';
import { useNavigate } from 'react-router-dom';

import Container from '../Container/Container';
import Button from '../Button/Button';
import styles from './PopularProducts.module.scss';
import PopularSwiper from './PopularSwiper/PopularSwiper';

export default function PopularProducts() {
  const navigate = useNavigate();

  return (
    <Container>
      <section className={styles.section_popular}>
        <div className={styles.popular_header}>
          <h2 className={styles.popular_headerTitle}>Popular Products</h2>
          <p className={styles.popular_headerText}>
            Medical marijuana can be beneficial in treating anxiety disorders
            and post-traumatic stress syndrome. Marijuana can help patients with
            depression and other mood-related conditions
          </p>
        </div>

        <PopularSwiper />

        <Button
          text="All Products"
          className="orangeBtn"
          onClick={() => {
            navigate('/shop');
          }}
        />
      </section>
    </Container>
  );
}
