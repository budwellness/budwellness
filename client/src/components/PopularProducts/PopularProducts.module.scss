.section_popular {
  min-width: 320px;

  @include _320 {
    padding: 60px 0;
  }
  @include _768 {
    padding: 90px 0;
  }
  @include _1380 {
    padding: 160px 0;
  }

  .container {
    // max-width: 1380px;
    margin: 0 auto;
    padding-right: 0.75rem;
    padding-left: 0.75rem;

    @include _1380 {
      max-width: calc(1380px - 1.5rem);
    }

    .popular_header {
      text-align: center;
      margin-bottom: 30px;

      @include _768 {
        margin-bottom: 40px;
      }

      .popular_headerTitle {
        margin: 0 0 15px;
        color: $main-green;
        font-family: $font-nunito;
        font-weight: 800;
        font-size: 30px;
        line-height: 1.3;

        @include _1380 {
          font-size: 36px;
          line-height: 1.2;
        }
      }

      .popular_headerText {
        max-width: 500px;
        margin: 0 auto;
        font-size: 17px;
        line-height: 25.5px;
      }
    }

    button {
      display: block;
      margin: 0 auto;
    }
  }
}
