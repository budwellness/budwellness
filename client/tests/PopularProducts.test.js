/* eslint-disable */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import PopularProducts from '../src/components/PopularProducts/PopularProducts';

jest.mock(
  '../src/components/PopularProducts/PopularSwiper/PopularSwiper',
  () => () => <div>PopularSwiper</div>
);
jest.mock('../src/components/Button/Button', () => () => (
  <button>Button</button>
));

describe('PopularProducts component Smoke test (Snapshot test)', () => {
  test('should PopularProducts render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <PopularProducts />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
