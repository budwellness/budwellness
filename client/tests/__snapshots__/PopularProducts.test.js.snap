// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularProducts component Smoke test (Snapshot test) should PopularProducts render 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <section
      class="section_popular"
    >
      <div
        class="popular_header"
      >
        <h2
          class="popular_headerTitle"
        >
          Popular Products
        </h2>
        <p
          class="popular_headerText"
        >
          Medical marijuana can be beneficial in treating anxiety disorders and post-traumatic stress syndrome. Marijuana can help patients with depression and other mood-related conditions.
        </p>
      </div>
      <div>
        PopularSwiper
      </div>
      <button>
        Button
      </button>
    </section>
  </div>
</DocumentFragment>
`;
